### ISSUE TITLE
Feature: Create and send tabs to Tab Groups

### ISSUE THREAD

**Issue by alex-gunning** [philc/vimium#3541]
Tab Groups have recently been implemented in Chrome. It would be cool to be able to hotkey to create Tab Groups and send tabs to tab groups (perhaps via fuzzy selection)

![image](https://user-images.githubusercontent.com/12812053/79673294-d124f280-81d8-11ea-9938-9e7a3fdee81c.png)

Side note: This is accessible via a flag that is meant to be toggled-on in Chrome. I don't know if this is accessible via Chrome extensions yet. I also have no idea what the state of this is in Firefox.

**Comment by alex-gunning** [comment]
Also happy to work on this :) Just need to know if it's possible. I was thinking of contacting the people maintainers over at [Chromium](https://github.com/chromium/chromium)

**Comment by gdh1995** [comment]
It seems impossible because I didn't find any related apis on https://developer.chrome.com/extensions/tabs just now



---Original---

**Comment by alex-gunning** [comment]
Yeah, aggg. I guess I'll have to wait until it becomes supported.

**Comment by balloox** [comment]
Any update guys? I saw 2 apis:
- https://developer.chrome.com/extensions/tabs#method-group
- https://developer.chrome.com/extensions/tabs#method-ungroup

**Comment by alex-gunning-jumo** [comment]
@balloox Awesome! Thanks for the update. You have given me much joy today.

**Comment by balloox** [comment]
@alex-gunning-jumo, are you going to implement the feature? Oops just realized you're the author lmao :)

**Comment by alex-gunning-jumo** [comment]
@balloox Yes, this feature would be great. Everything nice and organised. I'll have to see about my time though, I'm a little pressed for it right now.

**Comment by alex-gunning-jumo** [comment]
I'm really excited to do this. I'll code it up in the first few days following new years if nobody else is doing it already.

**Comment by balloox** [comment]
@alex-gunning-jumo,  thanks man, really appreciate it. I'm stuck too tightly with cyberpunk, don't have so much time to do other things :)

**Comment by alex-gunning-jumo** [comment]
@balloox I've checked it out but this may actually turn out to be a significant amount of work.

The api which you linked above requires that we be on the 
```
  "manifest_version": 3
```

We are currently on version 2. 

We would need to first follow this migration guide [https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/)

and then fix all broken code.

Also, manifest v3 will only go live in [Chrome 88](https://www.xda-developers.com/manifest-v3-changes-browser-extensions-go-live-google-chrome-88/#:~:text=Availability%20and%20Rollout%20for%20Manifest,88%20reaches%20the%20stable%20branch.), so we'll be stuck using a dev channel until then. We obviously can't ship that to the general public.


Slightly off topic: I think there's some merit in prepping the codebase for this update and starting now couldn't hurt.

**Comment by alex-gunning-jumo** [comment]
An update. Chrome is releasing version 88 on the 19th of January.

This will enable the above. We could start a new fork then and begin the migration. I am still happy to do this.

**Comment by balloox** [comment]
@alex-gunning-jumo, will it cost a lot of time? I mean is this big work? I have zero knowledge about this (vimium, extension), is it worth to do this now or maybe later?

**Comment by balloox** [comment]
@alex-gunning-jumo, sorry for the misunderstanding but I have nearly zero knowledge about Vimium, Extension in general. Is it worth it to do this now or maybe later? Have a glanced at the migration guide and I think it may cost a lot of time. Once again I'm really sorry.

**Comment by alex-gunning-jumo** [comment]
@balloox I have only basic knowledge of the codebase (fixed another [minor bug](https://github.com/philc/vimium/pull/3584) which I'm not sure will ever be merged.)

I'm not sure exactly how much time but I'm going to give it a bash anyway.
